ChattoAdditions/Source/Input/Media/MediaCell/MediaInputDataProvider.swift

```diff
@@ -155,39 +163,57 @@ final class PhotosInputDataProvider: NSObject, PhotosInputDataProviderProtocol,
             }
             var requestId: Int32 = -1
             self.fullImageRequests[asset] = request
-            requestId = imageManager.requestImageData(for: asset, options: options, resultHandler: { [weak self] (data, _, _, info) in
-                guard let sSelf = self else { return }
-                let result: PhotosInputDataProviderResult
-                if let data = data, let image = UIImage(data: data) {
-                    result = .success(image)
-                } else {
-                    result = .error(info?[PHImageErrorKey] as? Error)
-                }
-                request.handleCompletion(with: result)
-                sSelf.fullImageRequests[asset] = nil
-            })
+
+            if asset.mediaType == .image {
+                requestId = imageManager.requestImageData(for: asset, options: options, resultHandler: { [weak self] (data, _, _, info) in
+                    guard let sSelf = self else { return }
+                    let result: MediaInputDataProviderResult
+                    if let data = data, let image = UIImage(data: data) {
+                        result = .successImage(image)
+                    } else {
+                        result = .error(info?[PHImageErrorKey] as? Error)
+                    }
+                    request.handleCompletion(with: result)
+                    sSelf.fullImageRequests[asset] = nil
+                })
+            } else if asset.mediaType == .video {
+                let videoOptions = PHVideoRequestOptions()
+                videoOptions.deliveryMode = .fastFormat
+                requestId = imageManager.requestAVAsset(forVideo: asset, options: videoOptions, resultHandler: { [weak self] (avAsset, _, info) in
+                    let result: MediaInputDataProviderResult
+                    if let data = avAsset as? AVURLAsset {
+                        result = .successVideo(data.url)
+                    } else {
+                        result = .error(info?[PHImageErrorKey] as? Error)
+                    }
+                    DispatchQueue.main.async {
```

Comment:
Why? Could you leave a comment, please? It's not obvious why do you need this call.

Reply:
because completion comes from background thread, will add comment